src/core/service.c

```diff
@@ -381,7 +381,7 @@ static int service_add_fd_store(Service *s, int fd, const char *name) {
                 if (r > 0) {
                         /* Already included */
                         safe_close(fd);
-                        return 1;
+                        return 0;
```

Comment:
Actually `service_add_fd_store` never returned 0, it always took possession of the fd (or returned an error). With my change, it still always takes possession, but now 0 means it was an old fd, and 1 means it was a new fd. I can either add a comment or restore previous. Preference?